code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/NotificationBuilder.kt

```diff
@@ -67,9 +83,175 @@ object NotificationBuilder {
+        return getNotificationBuilder(context, notificationData, trackerActivityClass, broadcastReceiverClass)
+    }
+
+    /**
+     * Constructs a [NotificationCompat.Builder] object from the provided [intent]
+     *
+     * @param intent [Intent] containing the data needed for the notification construction
+     * @param trackerActivityClass [Class] of the [Activity] to be launched when the notification is clicked
+     * @param broadcastReceiverClass [Class] of the [BroadcastReceiver] to be used for handling notification actions
+     * @return [NotificationCompat.Builder] object
+     * @throws [NotificationConstructionFailedException] if the notification construction fails due to missing data
+     * @throws [IllegalArgumentException] if the provided message data has invalid data
+     */
+    @Throws(NotificationConstructionFailedException::class, IllegalArgumentException::class)
+    @JvmStatic
+    fun constructNotificationBuilder(
+        intent: Intent,
+        trackerActivityClass: Class<out Activity>?,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?
+    ): NotificationCompat.Builder {
+        val context = ServiceProvider.getInstance().appContextService.applicationContext
+            ?: throw NotificationConstructionFailedException("Application context is null, cannot build a notification.")
+        val extras = intent.extras ?: throw NotificationConstructionFailedException("Intent extras are null, cannot re-build the notification.")
+        val intentData = IntentData(extras, intent.action)
+        return getNotificationBuilder(context, intentData, trackerActivityClass, broadcastReceiverClass)
+    }
+
+    /**
+     * Handles the remind later intent by scheduling a [PendingIntent] to the [broadcastReceiverClass]
+     * which will be fire at a time specified in the [remindLaterIntent].
+     *
+     * Once the PendingIntent is fired, the [broadcastReceiverClass] is responsible for
+     * reconstructing the notification and displaying it .
+     *
+     * @param remindLaterIntent [Intent] containing the data needed to schedule and recreate the notification
+     * @param broadcastReceiverClass [Class] of the [BroadcastReceiver] that will be fired when the [PendingIntent] resolves at a later time
+     */
+    @Throws(NotificationConstructionFailedException::class)
+    @JvmStatic
+    fun handleRemindIntent(
+        remindLaterIntent: Intent,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?
+    ) {
+        val context = ServiceProvider.getInstance().appContextService.applicationContext
+            ?: throw NotificationConstructionFailedException("Application context is null, cannot schedule notification for later.")
+        if (broadcastReceiverClass == null) {
+            Log.trace(
+                LOG_TAG,
+                SELF_TAG,
+                "Broadcast receiver class is null, will not schedule a notification from the received" +
```

Comment:
nit: update the logs to use interpolation `intent with action $remindLaterIntent.action`